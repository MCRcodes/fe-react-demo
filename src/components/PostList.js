import React, { useState } from "react";
import PropTypes from "prop-types";

import Post from "./Post";

const PostList = ({ posts }) => {
  const [lastUpvoted, setLastUpvoted] = useState("");

  const handleUpvote = (title) => {
    setLastUpvoted(title);
  };

  return (
    <>
      {lastUpvoted ?? <div className="last-upvoted">{lastUpvoted}</div>}
      <div className="postlist">
        {posts.map((post) => (
          <Post key={post.id} postData={post} handleUpvote={handleUpvote} />
        ))}
      </div>
    </>
  );
};

PostList.propTypes = {
  posts: PropTypes.arrayOf(
    PropTypes.shape({
      author: PropTypes.string,
      body: PropTypes.string,
      date: PropTypes.string,
      isPublished: PropTypes.bool,
      tags: PropTypes.arrayOf(PropTypes.string),
      title: PropTypes.string,
    })
  ).isRequired,
};

export default PostList;
