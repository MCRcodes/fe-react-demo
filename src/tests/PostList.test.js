import React from "react";
import renderer from "react-test-renderer";
import PostList from "../components/PostList";

describe("PostList", () => {
  const validProps = {
    posts: [
      {
        id: 1,
        tags: [],
      },
      {
        id: 2,
        tags: [],
      },
    ],
  };

  test("renders as expected", () => {
    const testRenderer = renderer
      .create(<PostList posts={validProps.posts} />)
      .toJSON();

    expect(testRenderer).toMatchSnapshot();
  });
});