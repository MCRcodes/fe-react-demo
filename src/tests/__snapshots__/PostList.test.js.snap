// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList renders as expected 1`] = `
Array [
  "",
  <div
    className="postlist"
  >
    <div
      className="post"
    >
      <div
        className="post-body"
      >
        <h2 />
        <p>
          Coming soon!
        </p>
      </div>
      <div
        className="post-counter"
      >
        <span
          style={
            Object {
              "marginRight": "12px",
            }
          }
        >
          Total votes: 
          0
        </span>
        <button
          onClick={[Function]}
          type="button"
        >
          Upvote this
        </button>
      </div>
      <div
        className="post-author"
      >
        Author: 
      </div>
      <div
        className="post-date"
      >
        Published: 
      </div>
      <h3>
        Tags:
      </h3>
      <ul
        className="post-tags"
      />
    </div>
    <div
      className="post"
    >
      <div
        className="post-body"
      >
        <h2 />
        <p>
          Coming soon!
        </p>
      </div>
      <div
        className="post-counter"
      >
        <span
          style={
            Object {
              "marginRight": "12px",
            }
          }
        >
          Total votes: 
          0
        </span>
        <button
          onClick={[Function]}
          type="button"
        >
          Upvote this
        </button>
      </div>
      <div
        className="post-author"
      >
        Author: 
      </div>
      <div
        className="post-date"
      >
        Published: 
      </div>
      <h3>
        Tags:
      </h3>
      <ul
        className="post-tags"
      />
    </div>
  </div>,
]
`;
